docs/App Authenticator.md

```diff
@@ -0,0 +1,307 @@
+```
+
+**URL Parameters**
+
+| Key       | Example Value   | Description                          |
+| --------- | --------------- | ------------------------------------ |
+| tab_id    | RW_LDHk5zB8     | Tab Id of the User's Browser Session |
+| client_id | account-console | Client ID the User is logging in     |
+| key       | _see below_     | Keycloak Action Token as JWT         |
+
+**Decoded Keycloak Action Token**
+
+**Header**
+
+```json
+{
+	"alg": "HS256",
+	"typ": "JWT",
+	"kid": "1aa3caa1-4fa0-4355-aa5d-ee5a7789044f"
+}
+```
+
+**Payload**
+
+```json
+{
+	"exp": 1701428464,
+	"iat": 1701428164,
+	"jti": "e9aea2c4-9c87-420d-8667-684c09234e07",
+	"iss": "http://192.168.2.127:8080/realms/dev",
+	"aud": "http://192.168.2.127:8080/realms/dev",
+	"sub": "b5bf6396-21cd-466c-9633-0de3a152ba63",
+	"typ": "app-setup-action-token",
+	"nonce": "e9aea2c4-9c87-420d-8667-684c09234e07",
+	"oasid": "0a18a341-5ea7-4ef8-bdc5-de7f469272b1.RW_LDHk5zB8.29595bfa-758e-421b-a8d3-0afb48d1728d"
+}
+```
+
+**Signature**
+
+```plain
+XLh09uLq9Ybx_fCIcMhWcNELy9wnPaGMZ8pRusJ_b_g
+```
+
+### Error Responses
+
+**About HTTP Status Codes**
+
+-   `4xx` indicate a problem caused by the client.
+-   `401 Unauthorized` Is returned when the client failed to prove it's identity. E.g. missing authentication information, invalid credentials or a presented JWT that failed verification.
+-   `403 Forbidden` is returned when the client proved it's identity but is not allowed to access the resource because of missing privileges.
+-   `5xx`indicate a server sided problem. Error codes from this range **MAY NOT** be returned if the problem was caused by the client. Any endpoint can return with this status code.
+
+**Error Object**
+
+Any error response can optionally return an error object as JSON in the body with the following shape. Since it is not possible to return a customized error on the action token endpoint, only the HTTP status code is returned without a body.
+
+```json
+{
+	"error": "some_error_type",
+	"message": "details about the occurred error"
+}
+```
+
+### Authentication
+
+**Signature Tokens**
+
+The API endpoints require an authentication mechanism that leverages client-side generated keypairs. While drafts for HTTP Message Signatures exist, they are no well-established standards.
+[draft-ietf-httpbis-message-signatures](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures).
+
+Instead of implementing concepts from this unfinished draft, the API uses client-side generated JSON Web Tokens (JWTs) as a form of request signature. The JWTs are signed using a private key stored on the client and transmitted in the `x-signature` header.
+
+The client should use the authenticator id as the `kid` claim and use an asymmetric signature algorithm. The acceptable algorithms are:
+
+-   `PS512` with an `RSASSA-PSS` asymmetric key
+-   `ES512` with an `EC` asymmetric key
```

Comment:
I think thats fine. I want to stick close to the on board Keycloak JWT toolset and Keycloak seems to provide signature verifiers for your proposed algorithms: https://github.com/keycloak/keycloak/tree/main/services/src/main/java/org/keycloak/crypto